File: PersonnelDepartment/Domain/Departments/Department.cs
namespace PersonnelDepartment.Domain.Departments;

public class Department
{
    public Guid Id { get; }
    public String Name { get; }
    public String PhoneNumber { get; }

    public Department(Guid id, String name, String phoneNumber)
    {
        Id = id;
        Name = name;
        PhoneNumber = phoneNumber;
    }
}

File: PersonnelDepartment/ClientApp/src/domain/departments/department.ts
export class Department {
    constructor(
        public id: string,
        public name: string,
        public phoneNumber: string
    ) { }
}

export function mapToDepartment(data: any): Department {
    return new Department(data.id, data.name, data.phoneNumber);
}

export function mapToDepartments(data: any[]): Department[] {
    return data.map(mapToDepartment);
}